### ISSUE TITLE
Tried to compile locally on Windows 7 but filed

### ISSUE THREAD

**Issue by AndrewCh** [kaitai-io/kaitai_struct_webide#43]
Setup: 
Windows 64 bit, 
Python 2.7.13 |Anaconda 4.4.0 (64-bit)| (default, May 11 2017, 13:17:26) [MSC v.
1500 64 bit (AMD64)] on win32
node: v6.11.3
tried typescript: 2.5.2

I've followed the steps to compile web ide locally on Windows 7 but failed due to this error.
`C:\Users\test\kaitai_struct_webide>python serve.py --compile
auto compile started
please use 127.0.0.1:8888 on Windows (using localhost makes 1sec delay)
Changed file: ./src\worker\TemplateCompiler.ts
'tsc' is not recognized as an internal or external command,
operable program or batch file.
Compile errors:
 -`
1. Obviously, the typescript compiler was not found but description does not mention there is a need to install typescript separately.
2. After installing typescript manually with 
`C:\Users\test\kaitai_struct_webide>npm install -g typescript`
there were a bunch of weird errors during compillation process.
3. In addition, there is is a bug in genKaitaiFsFiles.py file since when  executing command below from the steps there is an error.
`C:\Users\test\kaitai_struct_webide>python serve.py --compile
auto compile started
Traceback (most recent call last):
  File "serve.py", line 111, in <module>
    import genKaitaiFsFiles
  File "C:\Users\iscamc\kaitai_struct_webide\genKaitaiFsFiles.py", line 21, in <
module>
    with open(outDir + 'js/kaitaiFsFiles.js', 'wt') as f:
IOError: [Errno 2] No such file or directory: '--compile/js/kaitaiFsFiles.js'
`
I was able to fix it changing in genKaitaiFsFiles.py
`outDir = sys.argv[1] + '/' if len(sys.argv) > 1 else ''`
to
`outDir = sys.argv[1] + '/' if len(sys.argv) > 2 else ''`

Any thoughts how to resolve this?

**Comment by koczkatamas** [comment]
Hey! Thanks for the report.

I fixed some issues (hopefully all) in commit https://github.com/kaitai-io/kaitai_struct_webide/commit/a4d13d4a579277b329d9bfbd8668312d0453fb01

I use VS Code for development and haven't tested the `--compile` command lately.

1) I fixed the compilation command by using the `tsc` version installed locally by `npm install`. (As typescript changes from version to version, which can break compilation, it's better not to use the global one. I am using TS 2.5.2 now btw).

2) Yeah, I think it was because the previous `tsc` command line tried to set up the compilation process manually. Now it will use the options specificed in `tsconfig.json`.

3) I fixed `genKaitaiFsFiles` too. I lazily just imported it as a module from `serve.py`. Now I fixed that by converting `genKaitaiFsFiles` to a module actually and make it possible to use as a separate app too.

To compilation instructions in `README.md` should work now (recursive clone, npm install, serve.py --compile).
